xbmc/filesystem/IFile.cpp

```diff
@@ -78,6 +78,27 @@ bool IFile::ReadString(char *szLine, int iLineLength)
   return true;
 }
 
+ssize_t IFile::GetMaxReadSize()
+{
+  int chunkSize = GetChunkSize();
+
+  switch(chunkSize)
+  {
+    case 0:
+    case 1:
+      // Backwards compatibility:
+      // CFileCache used to interpret a chunk size of 0 or 1 as
+      // 1) Get chunk size from advanced settings
+      // 2) Read one chunk at a time (the only way it knew how!)
```

Comment:
Don't you mean
```c
/// 0) Foo
/// 1) Bar
```
?

Reply:
No, it was the same two parts to the handling for both chunk size 0 and 1. But I see how that might be confusing.

Reply:
Clearer?
```
    case 0:
    case 1:
      // Backwards compatibility:
      // CFileCache used to handle a chunk size of 0 or 1 by
      // a) Getting the chunk size from advanced settings
      // b) Reading one chunk at a time (the only way it knew how!)
      // To not change this behavior for IFile implementations,
      // Assume the ONE_CHUNK behavior in those cases and let the IFiles
      // that want something different override this method.
      return SpecialReadSizeRequest::ONE_CHUNK;
```

Reply:
Much better! :smiley: